### PR TITLE
BUG: Fix inaccurate rolling.var calculation

### LINKED ISSUE #18430: BUG: rolling.corr() produces wrong result with equal values
#### Code Sample, a copy-pastable example if possible

```python
s = pd.Series([1,1,2,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,5,0,0,0,7,0,0,0])
pd.rolling_corr(s,s,6)

```
#### Problem description

rolling_corr is producing the wrong result:
```
python
pd.rolling_corr(s,s,6)
```
0     NaN
1     NaN
2     NaN
3     NaN
4     NaN
5     1.0
6     1.0
7     1.0
8     1.0
9     0.0
10    0.0
11    0.0
12    0.0
13    0.0
14    0.0
15    0.0
16    0.0
17    0.0
18    0.0
19    0.0
20    0.0
21    0.0
22    0.0
23    0.0
24    0.0
25    0.0
26    1.0
27    1.0
28    1.0
29    1.0
30    1.0
31    1.0
32    1.0
33    1.0

This should have nan's instead of 0's for windows with static data.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.3.final.0
python-bits: 64
OS: Linux
OS-release: 4.1.35-pv-ts2
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.20.3
pytest: 3.0.7
pip: 9.0.1
setuptools: 36.4.0
Cython: 0.25.2
numpy: 1.11.3
scipy: 0.19.0
xarray: None
IPython: 5.3.0
sphinx: 1.6.2
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: None
tables: 3.3.0
numexpr: 2.6.2
feather: None
matplotlib: 1.5.1
openpyxl: 2.4.8
xlrd: 1.0.0
xlwt: None
xlsxwriter: 0.9.8
lxml: None
bs4: 4.5.3
html5lib: 0.999
sqlalchemy: 1.1.10
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by Licht-T**
- [x] closes #18430
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/test_window.py [hunk 1]
```diff
@@ -2482,6 +2482,13 @@ def test_rolling_corr_pairwise(self):
         self._check_pairwise_moment('rolling', 'corr', window=10,
                                     min_periods=5)
 
+    def test_rolling_corr_with_zero_variance(self):
+        # GH 18430
+        s = pd.Series(np.zeros(20))
+        other = pd.Series(np.arange(20))
+
+        assert s.rolling(window=5).corr(other=other).isna().all()
```
Comment (jreback) [inline pandas/tests/test_window.py]: and for window 0 thru 6

**Review by jreback — CHANGES_REQUESTED** [review]
add a whatsnew note 0.21.1 bug fix

File: pandas/_libs/window.pyx [hunk 2]
```diff
@@ -675,9 +675,9 @@ cdef inline void remove_var(double val, double *nobs, double *mean_x,
     if val == val:
         nobs[0] = nobs[0] - 1
         if nobs[0]:
-            delta = (val - mean_x[0])
+            delta = val - mean_x[0]
             mean_x[0] = mean_x[0] - delta / nobs[0]
-            ssqdm_x[0] = ssqdm_x[0] - delta * (val - mean_x[0])
```
Comment (jreback) [inline pandas/_libs/window.pyx]: add sone commentary that u r using welford online algos- add a link here

**Comment by jreback** [comment]
@byospe can you have a look.

**Comment by Licht-T** [comment]
@jreback Fixed!

**Comment by jreback** [comment]
thanks @Licht-T can you rebase once again.

**Comment by Licht-T** [comment]
@jreback Rebased.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @Licht-T keep em coming!

**Pull request merged**